src/Repository.js

```diff
@@ -43,7 +44,34 @@ export default class Repository {
     return this.lernaJson && this.lernaJson.bootstrapConfig || {};
   }
 
+  get packages() {
+    if (!this._packages) {
+      this.buildPackageGraph();
+    }
+    return this._packages;
+  }
+
+  get filteredPackages() {
+    if (!this._filteredPackages) {
+      this.buildPackageGraph();
+    }
+    return this._filteredPackages;
+  }
+
+  get packageGraph() {
+    if (!this._packageGraph) {
+      this.buildPackageGraph();
```

Comment:
Honestly it was done like this to match asini's PR (which originated from here), so I'd be happy for it to be either way